src/bank.ts

```diff
@@ -0,0 +1,289 @@
+import { Rabbit } from './types';
+import { v4 as uuid } from 'uuid';
+import ResourceNotFoundError from './errors/resource-not-found-error';
+import InvalidRequestError from './errors/invalid-request-error';
+import ResourceExistsError from './errors/resource-exists';
+
+type Request = {
+  withdrawal: 'WithdrawalIdNotFound' | 'WithdrawalNotOnPendingState'
+} & {
+  account: 'AccountNotExists' | 'AccountMemberLevelNotExists' | "HighPointMemberlLevelNotExists" | 'AmountIsGreaterThanCurrentAmount'
+    | 'DailyWithdrawalExceed'
+} & {
+  memberLevel: 'MemberLevelNotExists' | 'PaymentMethodAndMemberLevelAlreadyExists'
+} & {
+  deposit: 'DepositNotExists' | 'DepositStatusNotFulfilled'
+} & {
+  id: 'PaymentMethodNotFound'
+};
+let workers: any[];
+export async function start(rabbit: Rabbit) {
+  workers = await Promise.all([
+    rabbit.createWorker('Bank.Query',
+     async function handleCommand({type, data}: {type: string, data: Request}) {
+      if (type === 'CanWithdraw') {
+        if (data.account === 'AccountMemberLevelNotExists') {
+          throw new InvalidRequestError(
+            'No member level being assigned to the account',
+            {
+              noMemberLevel: true,
+            }
+          );
+        }
+        
+        if (data.account === 'HighPointMemberlLevelNotExists') {
+          throw new InvalidRequestError(
+            'No member level being assigned to the account',
+            {
+              noMemberLevel: true,
+            }
+          );
+        }
+
+        if (data.account === 'AmountIsGreaterThanCurrentAmount') {
+          throw new InvalidRequestError('Withdrawal amount is not in range.', {
+            amount: 123.2,
+            minimumWithdrawal: 120.2,
+            maximumWithdrawal: 123.3,
+          });
+        }
+
+        if (data.account === 'DailyWithdrawalExceed') {
+          throw new InvalidRequestError('Maximum daily withdrawal exceed', {
+            withdrawals: 200.2,
+            maximum:300.5,
+          });
+        }
+        return new Promise(() => {
```

Comment:
No need to return a new Promise,. this is an async function,. Async function will always return promise.